### PR TITLE
Feature/massmint upload media zip

### LINKED ISSUE #5178: Mass Mint: step 2 - upload pictures
* #4964 

## Figma
---
- [Detailed interactions and logic is available Figma](https://www.figma.com/file/HCmIzbN0a2XhsOstHkLiwi/Mass-Mint---Handoff?node-id=0%3A1&t=ynDINS3WQGGG9Dj0-1). 
- all the needed screens
- [interactive prototype ](https://www.figma.com/proto/HCmIzbN0a2XhsOstHkLiwi/Mass-Mint---Handoff?page-id=0%3A1&node-id=1%3A332&viewport=1014%2C587%2C0.04&scaling=scale-down-width&starting-point-node-id=1%3A332&show-proto-sidebar=1)

Please take the prototype with a grain of salt, not everything is perfect and 1:1 to final design how it should look, so in case you don't know something, don't hesitate to ask @VladoTheBoi 

## STEP 2: UPLOAD PICTURES
- **supported files:** .zip
- **Supported art formats:** BMP, GIF, JPEG, PNG, SVG, TIFF, WEBP, MP4, OGV, QUICKTIME, WEBM, GLB, FLAC, MP3, JSON
- user can upload a .zip file containing the art
- user can either: 
  - tap and choose a files (tapping opens default OS finder)
  - drag and drop the file into the field
- uploading has an animation 
- once uploaded, user can still make a change by pressing "Choose File" button which opens the finder again and then choose another .zip file

---
image

### PR DESCRIPTION
**Pull request by daiagi**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).

👇 \_ Let's make a quick check before the contribution.

## PR Type

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

## Context

- [x] Closes #5178
- [ ] Requires deployment <>

#### Before submitting pull request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [x] I've posted a screenshot of demonstrated change in this PR

#### Optional

- [ ] I've tested it at </bsx/collection>
- [ ] I've tested PR on mobile
- [ ] I've written unit tests 🧪
- [ ] I've found edge cases

#### Had issue bounty label?

- [x] Fill up your KSM address: [Payout](https://beta.kodadot.xyz/transfer/?target=EfmnRhHaQqfT3phm4cUCHCU3gFVDoSPR1U9WXzMRQBMqZ4L)

#### Community participation

- [ ] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

## Screenshot 📸

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

[Screencast from 17-04-23 21:04:07.webm](https://user-images.githubusercontent.com/22791238/232508269-449d1e52-bc9c-47a1-8813-60e18c9ba119.webm)

### <samp>🤖 Generated by Copilot at 7e3a207</samp>

> _Sing, O Muse, of the skillful coder who devised_
> _A wondrous feature for the NFT gallery, where many_
> _Artists and collectors gather to admire and trade_
> _The digital tokens of their creative works and passions._

### CODE REVIEW

File: components/massmint/uploadCompressedMedia/UploadCompressedMedia.vue [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+          emit('zipLoaded', {
+            validFiles: validFiles.value,
+            areAllFilesValid: allValid.value,
+          })
+        }
+      })
+    }
+
+    reader.onerror = () => {
+      console.error('Error reading zip file.')
```
Comment (Jarsen136) [inline components/massmint/uploadCompressedMedia/UploadCompressedMedia.vue:95]: pls use $consola

File: components/massmint/uploadCompressedMedia/UploadCompressedMedia.vue [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+  {
+    disabled: false,
+  }
+)
+
+const loading = ref(false)
+const showCheckmark = ref(false)
+
+const acceptedMediaFormatsString =
+  'BMP, GIF, JPEG, PNG, SVG, TIFF, WEBP, MP4, OGV, QUICKTIME, WEBM, GLB, FLAC, MP3, JSON'
```
Comment (Jarsen136) [inline components/massmint/uploadCompressedMedia/UploadCompressedMedia.vue:53]: This string could be generated auto from `validFormats` list at `components/massmint/uploadCompressedMedia/useZipValidator.ts`

Comment (daiagi) [reply components/massmint/uploadCompressedMedia/UploadCompressedMedia.vue:53]: True that

File: components/shared/DragDrop.vue [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    title: '',
+    icon: 'cloud-arrow-up',
+    subtitle: undefined,
+    disabled: false,
+  }
+)
+
+const selectedFile = ref<File | null>(null)
+const emit = defineEmits(['fileSelected'])
+const subtitle = computed(() => props.subtitle || $i18n.t('dropUpload'))
```
[the remark this answers is not in the document]
Comment (daiagi) [reply components/shared/DragDrop.vue:72]: That was my first attempt. It did not compile.
I didn't look too deep at why

File: components/shared/DragDrop.vue [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+.check-icon {
+  font-size: 2.5rem;
+}
+
+.limit-width {
+  max-width: 25rem;
+}
+</style>
+
+<style lang="scss">
```
Comment (roiLeo) [inline components/shared/DragDrop.vue:101]: multiple style?

File: components/massmint/uploadCompressedMedia/useZipValidator.ts [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+  const validFiles: FileObject[] = []
+  const warnings: WarningObject[] = []
+
+  for (const [name, entry] of Object.entries(entries)) {
+    if (entry.isDirectory) {
+      warnings.push({
+        name,
+        reason: 'is a directory',
+      })
+      continue
```
Comment (roiLeo) [inline components/massmint/uploadCompressedMedia/useZipValidator.ts:69]: hm.. isn't it smelly to use `continue`?

File: locales/en.json [hunk 6]
```diff
@@ -624,10 +625,9 @@
         "message": "The symbol which is attributed to the NFT: for an example, Kusama is labeled as KSM"
       },
       "create": "Create collection",
-      "drop": "Drop collection logo here or click to upload or simple paste image from clipboard. We support various media types (PNG, JPEG, GIF, SVG)",
-      "massmintDrop": "Drop your file here or click to upload.",
-      "formats": "Supported formats:",
-      "compressZip": "Compress your Art files into a .ZIP file."
+      "drop": "Drop collection logo here or click to upload or simple paste image from clipboard. We support various media types (PNG, JPEG, GIF, SVG)"
+      
+
```
Comment (roiLeo) [inline locales/en.json:630]:
```suggestion
```

File: locales/en.json [hunk 7]
```diff
@@ -1275,6 +1276,10 @@
       }
     },
     "uploadPics": "Upload Pictures",
+    "uploadzipTip": "Compress your Art files into a .ZIP file.",
+    "uploadDescriptionTip": "Upload your Description file.",
+
```
Comment (roiLeo) [inline locales/en.json:1281]:
```suggestion
```

File: components/shared/DragDrop.vue [hunk 8]
```diff
@@ -0,0 +1,109 @@
+<template>
+  <o-upload
```
Comment (roiLeo) [inline components/shared/DragDrop.vue:2]: next we could create `NeoDropUpload` component

**Review by roiLeo — APPROVED** [review]
No buefy 👌 
Composition api👌 

let it roll 🎢

File: components/massmint/uploadCompressedMedia/useZipValidator.ts (hunk 5 above)
Comment (daiagi) [reply components/massmint/uploadCompressedMedia/useZipValidator.ts:69]: Lol maybe. I'll reactor it out

File: components/shared/DragDrop.vue (hunk 4 above)
Comment (daiagi) [reply components/shared/DragDrop.vue:101]: Yes I needed one unscoped class to effect oruga component internal class

File: components/massmint/uploadCompressedMedia/UploadCompressedMedia.vue (hunk 2 above)
Comment (daiagi) [reply components/massmint/uploadCompressedMedia/UploadCompressedMedia.vue]: done

File: components/massmint/uploadCompressedMedia/UploadCompressedMedia.vue (hunk 1 above)
Comment (daiagi) [reply components/massmint/uploadCompressedMedia/UploadCompressedMedia.vue]: done

**Review by Jarsen136 — APPROVED** [review]

File: components/shared/DragDrop.vue (hunk 4 above)
Comment (daiagi) [reply components/shared/DragDrop.vue]: alright i removed the double style tag by using

**Comment by exezbcz** [comment]
you get this lovely tag 
image

**Comment by vikiival** [comment]
pay 50 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $50 @ 31.81 USD/KSM ~ 1.572 $KSM
🧗 [EfmnRhHaQqfT3phm4cUCHCU3gFVDoSPR1U9WXzMRQBMqZ4L](https://kusama.subscan.io/account/EfmnRhHaQqfT3phm4cUCHCU3gFVDoSPR1U9WXzMRQBMqZ4L)
🔗 [0x03891c31b62576be21faf8b45d123ae1afec6ebeaf795559e41e3c8ecfdea990](https://kusama.subscan.io/extrinsic/0x03891c31b62576be21faf8b45d123ae1afec6ebeaf795559e41e3c8ecfdea990)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**